vulture_os/toolkit/api_parser/trendmicro_visionone/trendmicro_visionone.py

```diff
@@ -0,0 +1,170 @@
+
+        result = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
+        if result.status_code != 200:
+            raise TrendmicroVisionOneAPIError(
+                f"Error on URL: {self.URL_BASE + url_path} Status: {result.status_code} Reason/Content: {result.content}")
+
+        return result.json()['items']
+
+    def _get_auditlogs(self, since=None, to=None):
+        url_path = '/v3.0/audit/logs'
+
+        query_params = {'startDateTime': since,
+                        'endDateTime': to,
+                        'orderBy': 'createdDateTime desc',
+                        'top': 50,
+                        'labels': 'all'}
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token,
+                   'Accept': 'application/json',
+                   'TMV1-Filter': ''}
+
+        r = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
+
+        if r.status_code != 200:
+            raise TrendmicroVisionOneAPIError(
+                f"Error on URL: {self.URL_BASE + url_path} Status: {r.status_code} Reason/Content: {r.content}")
+
+        return r.json()['items']
+
+    def _get_OAT(self, since=None, to=None):
+
+        url_path = '/v3.0/audit/logs'
+
+        query_params = {'detectedStartDateTime': since,
+                        'detectedEndDateTime': to,
+                        'ingestedStartDateTime': since,
+                        'ingestedEndDateTime': to,
+                        'top': 50}
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token,
+                   'Accept': 'application/json',
+                   'TMV1-Filter': ''}
+
+        r = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
+
+        if r.status_code != 200:
+            raise TrendmicroVisionOneAPIError(
+                f"Error on URL: {self.URL_BASE + url_path} Status: {r.status_code} Reason/Content: {r.content}")
+
+        return r.json()['items']
+
+    def _format_OAT_log(self, log):
+        return log["detail"]
+
+
+    def execute(self):
+
+        since = self.frontend.last_api_call or (timezone.now() - timedelta(hours=1))
+        to = timezone.now()
+        msg = f"Parser starting from {since} to {to}"
+        logger.info(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
+        start_time = since.isoformat().replace("+00:00", "Z")
+        end_time = to.isoformat().replace("+00:00", "Z")
+
+        result = self._get_alerts(start_time, end_time)
+
+        self.write_to_file(result)
+
+        result = self._get_auditlogs(start_time, end_time)
+
+        self.write_to_file(result)
+
+        result = self._get_OAT(start_time, end_time)
+
+        self.write_to_file(self._format_OAT_log(result))
+
+        # Writting may take some while, so refresh token in Redis
+        self.update_lock()
+
+        self.frontend.last_api_call = to
```

Comment:
Add try/except to catch raise exception in sub-function
And don't update "self.frontend.last_api_call = to" if exception raised